### PR TITLE
Vulnerabilites fixes

### PR DESCRIPTION
**Pull request by barbararochazup**

### CODE REVIEW

File: moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt [hunk 1]
```diff
@@ -33,8 +33,13 @@ class FindAllUsersInteractorImpl @Inject constructor(
...
 
-    override fun execute(name: String?, email: String?, pageRequest: PageRequest): ResourcePageResponse<UserResponse> {
-        return convert(userService.findAll(name, email, pageRequest))
+    override fun execute(name: String?, email: String?, authorization: String, pageRequest: PageRequest): ResourcePageResponse<UserResponse> {
+        val user = userService.findByToken(authorization)
+        if (user.root) {
+            return convert(userService.findAll(name, email, pageRequest))
+        }
+
+        return convert(userService.findAll(user.name, user.email, pageRequest))
```
Comment (pedroborgeszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt:42]: I think only user with root permission should have access to this resource

Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt]: Make sense! I made the change according to the suggestion

File: moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt [hunk 2]
```diff
@@ -33,8 +34,12 @@ class FindAllUsersInteractorImpl @Inject constructor(
...
 ) : FindAllUsersInteractor {
 
-    override fun execute(name: String?, email: String?, pageRequest: PageRequest): ResourcePageResponse<UserResponse> {
-        return convert(userService.findAll(name, email, pageRequest))
+    override fun execute(name: String?, email: String?, authorization: String, pageRequest: PageRequest): ResourcePageResponse<UserResponse> {
+        val user = userService.findByToken(authorization)
+        if (user.root) {
+            return convert(userService.findAll(name, email, pageRequest))
+        }
+        throw throw ForbiddenException()
```
Comment (pedroborgeszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt:42]: There is an extra "throw"

File: moove/legacy-moove/src/main/kotlin/io/charlescd/moove/legacy/moove/request/configuration/CreateRegistryConfigurationRequest.kt [hunk 3]
```diff
@@ -32,50 +32,44 @@ import com.fasterxml.jackson.annotation.JsonTypeInfo
...
 )
 
 data class CreateAzureRegistryConfigurationRequest(
     override val name: String,
     override val address: String,
-    override val authorId: String,
     val username: String,
     val password: String
-) : CreateRegistryConfigurationRequest(name, address, CreateRegistryConfigurationProvider.Azure, authorId)
+) : CreateRegistryConfigurationRequest(name, address, CreateRegistryConfigurationProvider.Azure)
```
Comment (pedroborgeszup) [inline moove/legacy-moove/src/main/kotlin/io/charlescd/moove/legacy/moove/request/configuration/CreateRegistryConfigurationRequest.kt:43]: I think these changes will affect the Villager module, didn't you need to do any modification there?

File: moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt (hunk 2 above)
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/user/impl/FindAllUsersInteractorImpl.kt:42]: extra  "throw" removed

File: moove/legacy-moove/src/main/kotlin/io/charlescd/moove/legacy/moove/request/configuration/CreateRegistryConfigurationRequest.kt (hunk 3 above)
Comment (barbararochazup) [reply moove/legacy-moove/src/main/kotlin/io/charlescd/moove/legacy/moove/request/configuration/CreateRegistryConfigurationRequest.kt:43]: I removed authorId only moove request, but internally this value (according with user in authorization token) continues to be sent to villager

**Review by pedroborgeszup — DISMISSED** [review]

**Review by luanecavalcantizup — APPROVED** [review]

**Review by monicaribeiro — CHANGES_REQUESTED** [review]

File: moove/application/src/main/kotlin/io/charlescd/moove/application/UserService.kt [hunk 4]
```diff
@@ -60,4 +61,36 @@ class UserService(private val userRepository: UserRepository) {
     fun save(user: User): User {
         return this.userRepository.save(user)
     }
+
+    fun delete(id: String) {
+        this.userRepository.delete(id)
+    }
+
+    fun findByToken(authorization: String): User {
```
[the remark this answers is not in the document]
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/UserService.kt]: feito

**Review by monicaribeiro — APPROVED** [review]

**Review by monicaribeiro — APPROVED** [review]

**Review by Leandroqo — DISMISSED** [review]

**Review by monicaribeiro — APPROVED** [review]

**Review by luanecavalcantizup — DISMISSED** [review]

**Review by monicaribeiro — DISMISSED** [review]

**Review by luanecavalcantizup — DISMISSED** [review]

**Review by Leandroqo — DISMISSED** [review]

**Pull request merged**